### ISSUE TITLE
Waterfall with show=False results in error plt not known.

### ISSUE THREAD

**Issue by sorenwacker** [shap/shap#2261]
Seems there is an import missing somewhere. The waterfall plot works, but not with `show=False`

```

shap.plots.waterfall(shap_values[0], show=False)


NameError                                 Traceback (most recent call last)
/tmp/ipykernel_3533873/1075320206.py in <module>
----> 1 shap.plots.waterfall(shap_values[0], show=False)
      2 savefig('waterfall.pdf')

/bulk/LSARP/envs/conda/pycaret/lib/python3.9/site-packages/shap/plots/_waterfall.py in waterfall(shap_values, max_display, show)
     41     # Turn off interactive plot
     42     if show is False:
---> 43         plt.ioff()
     44 
     45 
    NameError: name 'plt' is not defined

```

**Comment by kejlnz** [comment]
I have not tried it myself, but it seems it is just a typo:
Replacing all `plt` occurrences with `pl` in `_waterfall.py` should do the trick.

**Comment by sorenwacker** [comment]
Maybe ‘pyplot’ was imported as ‘pl’ instead of ‘plt’. Do you want me to create a pull request if that is the case? I am not sure what code standards you use. I would rather import it as ‘plt’ as it is commonly done.

**Comment by js850** [comment]
I hit this issue also, so I created a Pull Request to fix it as suggested by @kejlnz . #2342

**Comment by js850** [comment]
There are about 10 duplicate Pull Requests open all trying to fix this issue.  It would be wonderful if one of the maintainers would approve and merge one of them :)  @slundberg fyi

![image](https://user-images.githubusercontent.com/1587384/147825332-94192ac1-a3dd-46ec-933a-0aeca7c2cc33.png)

**Comment by rglynn** [comment]
@slundberg  can we please get this approved? It's been open for months now with multiple PRs trying to fix it.

**Comment by bello7777** [comment]
yes pls , same here it work fine locally, but on the cloud its 
Traceback (most recent call last):

 


  File "/home/appuser/venv/lib/python3.7/site-packages/shap/plots/_waterfall.py", line 337, in waterfall_legacy

    plt.ioff()

NameError: name 'plt' is not defined

**Comment by H2CO3** [comment]
I've hit the same problem. Why is it such a big deal to merge if multiple people have already suggested the (trivial) fix? It's very annoying, because the waterfall plot is the only visually reasonable way to visualize long feature names (as the JS version of the force plot can't rotate feature names, and the matplotlib version is so unstable and ugly it's not even funny).

@slundberg

**Comment by Mateko** [comment]
Same problem, fix please @slundberg

**Comment by Styx71** [comment]
I also have the same problem, changing plt's to pl inside _waterfall.py solved it on local. But I couldn't change syntax on docker container. Please solve it from source. @slundberg

**Comment by peete-keesel** [comment]
Same problem, please fix @slundberg 🙏

**Comment by petmoreno** [comment]
Same problem here as well, appreciate it if you fix @slundberg . Thanks in advance

**Comment by rherman9** [comment]
Same problem! Replacing `plt` occurence with `pl` in `_waterfall.py` fixes it.

**Comment by francescopisu** [comment]
This project is pretty dead, unfortunately.
You can try this if you like, the end result appears to be the same. The only difference is graphical.
```
exp = shap.Explainer(model, X)
sh = exp(X) # get shap values
sh.display_data = X_display.values # X_display has pretty names for display purposes

idx = 113
# you can specify custom ordering if you want to
feature_order = [...]
shap.plots.bar(sh[idx], show=False, order=feature_order)
```

**Comment by rluethy** [comment]
Ran into this error today. Is there a time frame for the fix to be merged?

**Comment by rglynn** [comment]
This appears to have been fixed in Slundberg's commit last week (https://github.com/slundberg/shap/commit/f7c0678f95c72981632198abc3bdc6572d599505

**Comment by simon-barley** [comment]
This was fixed in `shap==0.41.0`
;)
